ledger/apply/apply.go

```diff
@@ -18,21 +18,102 @@ package apply
 
 import (
 	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/crypto"
 	"github.com/algorand/go-algorand/data/basics"
 	"github.com/algorand/go-algorand/data/transactions"
 	"github.com/algorand/go-algorand/data/transactions/logic"
 )
 
+// AccountData is like basics.AccountData, but without any maps
+// containing AppParams, AppLocalState, AssetHolding, or AssetParams.
+type AccountData struct {
+	Status             basics.Status
+	MicroAlgos         basics.MicroAlgos
+	RewardsBase        uint64
+	RewardedMicroAlgos basics.MicroAlgos
+
+	VoteID          crypto.OneTimeSignatureVerifier
+	SelectionID     crypto.VRFVerifier
+	VoteFirstValid  basics.Round
+	VoteLastValid   basics.Round
+	VoteKeyDilution uint64
+
+	AuthAddr           basics.Address
+	TotalAppSchema     basics.StateSchema
+	TotalExtraAppPages uint32
+}
+
+// ToApplyAccountData returns apply.AccountData from basics.AccountData
+func ToApplyAccountData(acct basics.AccountData) AccountData {
+	return AccountData{
```

Comment:
I guess `apply.AccountData` is a temporary type while we do not have other/better types.
The idea is to have in-mem and persistent data types for components of `basics.AccountData`

Reply:
I get that, but it seems to me that this in-memory type is, in fact, the beginning of basics.AccountData, so why not make that explicit?

Reply:
There was a bigger potential change I considered, where basics.AccountData loses these maps altogether, ledger.Lookup() gets split into separate acct/app/asset methods too, all former users of AccountData are updated to get asset data separately, and the original AccountData type (and its msgp encoding) is moved inside ledger and becomes an implementation detail of account DB, eval, COW, etc. 

I didn't think of embedding it as an AccountBase in AccountData because I figured we this PR was moving us towards that goal, starting with apply.Balances.

However looking at what's missing, really only the REST API is left after this I believe.

Reply:
oh, also I was worried a change to basics.AccountData directly might involving updating a ton of tests.

Reply:
I see, so the plan might be that basics.AccountData becomes what I was calling basics.AccountBase, and it gets (maybe) embedded in ledger.accountFull.  The apply package will take the various pieces it needs, never an entire ledger.accountFull.

Reply:
y'know, I already changed cowForLogicLedger to have Get() return the new apply.AccountData, which is now out of sync with Ledger.Get() returning basics.AccountData, so I will try plumbing through Ledger.GetAppParams, etc to see how much more changes it would be.

Reply:
Ok, don't get carried away on my behalf.  This is nice enough to do as is, and I don't want it to get so large we we have trouble moving forward incrementally.

Reply:
I tried tracking down everywhere that was using the maps in basics.AccountData, it was the REST API (account data and dryrun endpoints), tealdbg, and goal, all calling through Ledger.Lookup() basically. I think it's probably best for now to have ledger.Ledger just hide the logic behind Lookup() of joining the asset/app data with account data, rather than change all the tests and callers to Lookup that use basics.AccountData.

Reply:
That's in: https://github.com/cce/go-algorand/compare/split-balances...split-balances-hide-accountdata-maps